### ISSUE TITLE
[Bug]: Extremely long time initializing Neo4jPropertyGraphStore for larger graphs

### ISSUE THREAD

**Issue by hypernovas** [run-llama/llama_index#16204]
### Bug Description

It takes about 14 min to initiate the graph store with 3558 entities. I feel this is because `refresh_schema() ` does not handle large graphs well. Maybe not using async? 

I pasted the log I ingested into the lib code. This is taking way too long if for things beyond experiments. Wondering  if you guys could help on this?

```
2024-09-24 17:22:40.552042 Step 1: Sanitize query output
2024-09-24 17:22:40.552075 Step 2: Enhanced schema
2024-09-24 17:22:40.552079 Step 3: Create driver
2024-09-24 17:22:40.552180 Step 4: Create async driver
2024-09-24 17:22:40.552232 Step 5: Set database
2024-09-24 17:22:40.552236 Step 6: Create structured schema
2024-09-24 17:22:40.552238 Step 7: Create supports vector index
2024-09-24 17:36:02.882747 Step 8: Create index
2024-09-24 17:36:02.888786 Step 9: Verify version
```

### Version

llama-index-graph-stores-neo4j==0.3.2, llama-index==0.11.11

### Steps to Reproduce

1. Use local Neo4j desktop server
2. Ingest more than 3000 entities with proper relationships (reasonable text description)
3. Load a Neo4j graph_store
```
graph = Neo4jPropertyGraphStore(
    username="neo4j", password="xxxxxx", url="bolt://localhost:7687", database="neo4j"
)
```

**Comment by kevin-yang-racap** [comment]
TLDR: Set neo4j_property_graph_store.allows_structured_queries=False (variable not parameter) to prevent it from refreshing_schema. This will disable you from using Text2Cypher retriever. If you only care about the start up and not every insertion refresh,  you can just set refresh_schema=False.

Looks like the Neo4j apoc proc meta.graphSample just takes a very long time to complete as the complexity of the graph scales up. I suspect this problem scales with number of entity/relation types you have because they use a statistical representation.


If you are using Neo4j, I would recommend setting refresh_schema=False when you are initializing the Neo4jPropertyGraphStore. This will prevent it from refreshing on startup. You should also set graph_store.allows_structured_queries=False as it natively sets it to True. This will prevent it from trying to refresh every insertion. Do note that this will prevent you from using Text2Cypher retriever.

If you still want to use Text2Cypher perhaps think about limiting the ontology of your KG.
